bundles/org.eclipse.swt/Eclipse SWT Drag and Drop/cocoa/org/eclipse/swt/dnd/DropTarget.java

```diff
@@ -115,30 +115,38 @@ public class DropTarget extends Widget {
 	static final String DEFAULT_DROP_TARGET_EFFECT = "DEFAULT_DROP_TARGET_EFFECT"; //$NON-NLS-1$
 	static final String IS_ACTIVE = "org.eclipse.swt.internal.isActive"; //$NON-NLS-1$
 
+public void handleLabels(Control c, long cls) {
+	if (c instanceof Label) {
+		long imageView = 0;
+		if ((imageView = OS.objc_msgSend(c.view.id, OS.sel_getImageView)) != 0) {
+			cls = OS.object_getClass(imageView);
+			addDragHandlers(cls);
+		}
+	} else if (c instanceof Composite) {
+		Control[] cal = ((Composite) c).getChildren();
+		for (Control child : cal) {
+			handleLabels(child, cls);
+		}
+	}
+}
 void addDragHandlers() {
 	// Our strategy here is to dynamically add methods to the control's class that are required
 	// by NSDraggingDestination. Then, when setTransfer is called, we just register
 	// the types with the Control's NSView and AppKit will call the methods in the protocol
 	// when a drag goes over the view.
-
 	long cls = OS.object_getClass(control.view.id);
 
 	if (cls == 0) {
 		DND.error(DND.ERROR_CANNOT_INIT_DROP);
 	}
 
+	// If the content view can be image view, then add the dragging methods to image view too.
```

Comment:
Looks like the comments are misplaced, please move it inside `handleLabels` (line 120).

Reply:
I have made the changes.